sopel/config/core_section.py

```diff
@@ -181,6 +185,15 @@ def homedir(self):
     capabilities.
     """
 
+    perform_commands = ListAttribute('perform_commands')
```

Comment:
While I'm not crazy about this attribute name, my own ideas aren't much better:

  - `startup_commands`
  - `connect_commands`
  - `do_on_connect`
  - … more lame ideas

But I definitely think the choice to use "perform" came from my ZNC analogy, which it probably shouldn't. The name of this config setting should make sense to someone who's ever even _heard_ of a ZNC.

Reply:
I went with `commands_on_connect`...
- `startup_commands` and `connect_commands` (despite being documented) connote that these commands will be executed _during_ the startup/connect, and not _after_
- `do_on_connect`... well... not bad, but I hope that `commands_on_connect` is slightly clearer.

Reply:
Didn't notice until today, but… The name of the setting should be changed, too.

Also, in other code, the old name is still used. This PR would fail tests if it had them. :laughing: